content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+              STATUS_OK = 'Ok'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
+                  @handle.root['ae_result'] = task.status == STATUS_OK ? 'ok' : 'error'
+                else
+                  @handle.root['ae_result'] = 'retry'
+                  @handle.root['ae_retry_interval'] = '1.minute'
+                end
+              end
+
+              def refresh_task
+                task_id = @handle.get_state_var(:refresh_task_id).try(:first)
```

Comment:
@lfu what is the order of the task_ids, could we check that both the tasks are done instead of just one. This code only seems to be testing for 1 task id.

Reply:
@lfu when we store the task ids we might want to store it as an array in the state vars and when we check we would have to validate that all the tasks have completed successfully.

Reply:
Processing as an array is a good idea but we could ensure it is an array here if the common use case is going to be a single task id.

Reply:
@gmcculloug @lfu 
The earlier code was doing this
      task_ids.each { |tid| MiqTask.wait_for_taskid(tid) } if sync
It was handling an array in https://github.com/ManageIQ/manageiq/blob/d992f980a857db789c8aa8b3636bf1b6559ca539/app/models/ems_event/automate.rb#L23

Reply:
I see now, you are dealing with multiple miq_task objects coming back from the backend provider call.  👍

Reply:
@mkanoor @gmcculloug It is true that the ```task_ids``` returned back from refresh is in an array. [Refresh](https://github.com/ManageIQ/manageiq/blob/962af5d7a934d8d286ac2f938a5d5e05c31efd8f/app/models/ems_refresh.rb#L56) would create multiple ```MiqTasks``` based on the number of ```ems``` these refresh targets belong to.  
In our case, the event belongs to one refresh target and it is for sure the refresh would create only one task.

Reply:
Other than returning ```task_ids``` in an array, the refresh may also return [```nil```](https://github.com/ManageIQ/manageiq/blob/962af5d7a934d8d286ac2f938a5d5e05c31efd8f/app/models/ems_refresh.rb#L56) when target is empty.